lib/fs/filesystem.go

```diff
@@ -16,6 +16,18 @@ import (
 	"time"
 )
 
+type FilesystemVariant int32
+
+const (
+	FilesystemVariantBasic FilesystemVariant = iota
+	FilesystemVariantMtime
+	FilesystemVariantCase
+	FilesystemVariantError
+	FilesystemVariantWalk
+	FilesystemVariantFake
+	FilesystemVariantLog
+)
+
```

Comment:
Currently this has overlap with `FilesystemType`, as in basic and fake exist there too. To make it clearly distinct from `FilesystemType`, this could be `FilesystemWrapperType` and basic and fake become just `FilesystemWrapperTypeNoWrapper` (or anything similar, naming being hard as always). Respectively basic and fake don't need a wrapper-type at all, as they wrap nothing, and just return false on their `underlying` and `variant` methods.

Reply:
Agree